apps/sodo-search/src/search-index.js

```diff
@@ -133,4 +136,55 @@ export default class SearchIndex {
             tags: this.#normalizeSearchResult(tags)
         };
     }
+
+    checkLocaleCjk(locale) {
+        if (!locale) {
+            return false;
+        }
+        // Chinese
+        if (locale === 'zh' || locale === 'zh-Hans' || locale === 'zh-Hant') {
+            return true;
+        }
+        // Japanese
+        if (locale === 'ja') {
+            return true;
+        }
+        // Korean
+        if (locale === 'ko') {
+            return true;
+        }
+        return false;
+    }
+
+    getCjkFlexsearchDocumentPatchOptions(locale) {
+        // https://github.com/nextapps-de/flexsearch?tab=readme-ov-file#cjk-word-break-chinese-japanese-korean
+
+        const isLocaleCjk = (() => {
+            if (!locale) {
+                return false;
+            }
+            // Chinese
+            if (locale === 'zh' || locale === 'zh-Hans' || locale === 'zh-Hant') {
+                return true;
+            }
+            // Japanese
+            if (locale === 'ja') {
+                return true;
+            }
+            // Korean
+            if (locale === 'ko') {
+                return true;
+            }
+            return false;
+        })();
+
+        if (!isLocaleCjk) {
+            return {};
+        }
+
+        return {
+            // eslint-disable-next-line no-control-regex
+            encode: str => str.replace(/[\x00-\x7F]/g, '').split('')
```

Comment:
This strips out most non-CJK characters.  (It leaves Cyrillic characters, the Bengali alphabet, and a subset of characters with diacritics.)  So implementing this would prevent indexing of nearly anything that isn't in CJK characters.  Although I recognize that you've carefully only applied this to sites with locale set to one of these character sets, I suspect many blogs have some Latin characters, such as people's names, foreign words, etc.  This would make them unsearchable.  Please see my other comment, that offers an encoding option that preserves multilingual text. :)